### PR TITLE
Issue-5541 Use Android 12 (API 31) as target API

### PR DESCRIPTION
**Pull request by AGulev**
Extender update for https://github.com/defold/defold/pull/6510

### CODE REVIEW

File: server/docker-base/Dockerfile [hunk 1]
```diff
@@ -283,7 +283,7 @@ RUN \
     chmod +w -R ${ANDROID_SDK_ROOT} && \
     chmod -R 755 ${ANDROID_ROOT}/android-ndk-r${ANDROID_NDK20_VERSION} && \
 # check that dx installed properly
-    ls -la ${ANDROID_SDK_BUILD_TOOLS_PATH}/dx && \
+    ls -la ${ANDROID_SDK_BUILD_TOOLS_PATH}/d8 && \
```
Comment (AGulev) [inline server/docker-base/Dockerfile:286]: This SDK doesn't have `dx` tool anymore, use `d8` to check if SDK installed.

File: server/src/test/java/com/defold/extender/AuthenticationTest.java [hunk 2]
```diff
@@ -31,7 +31,7 @@
 public class AuthenticationTest {
 
     private static final int EXTENDER_PORT = 9000;
-    private static final String SDK_VERSION = "d31d6397a72178541a5ef6e7ef2bed090d828f58"; // 1.2.190
+    private static final String SDK_VERSION = "d393bae6a361f86cf2263ab312c9b3cea45253ab"; // 1.2.191
```
Comment (AGulev) [inline server/src/test/java/com/defold/extender/AuthenticationTest.java:34]: I get "can't find SDK" with 1.2.190. 
So, I updated version and now tests are fine

File: server/src/test/java/com/defold/extender/IntegrationTest.java [hunk 3]
```diff
@@ -115,13 +115,9 @@ public String toString() {
                 // "a" is a made up sdk where we can more easily test build.yml fixes
                 new DefoldVersion("a", new Version(0, 0, 0), new String[] {"x86_64-osx", "armv7-android", "x86_64-win32"} ),
 
-                // At 1.2.178 we updated to Emscripten 2.x
-                new DefoldVersion("af6a29c2a1e2545e2d033790089c606ac9f0bb7a", new Version(1, 2, 178), new String[] {"armv7-android", "armv7-ios", "arm64-ios", "x86_64-osx", "x86_64-linux", "x86_64-win32", "js-web", "wasm-web"}),
-                new DefoldVersion("1f5712609c345f870b691a85d611d4825d22a718", new Version(1, 2, 184), new String[] {"armv7-android", "armv7-ios", "arm64-ios", "x86_64-osx", "x86_64-linux", "x86_64-win32", "js-web", "wasm-web"}),
```
Comment (AGulev) [inline server/src/test/java/com/defold/extender/IntegrationTest.java:120]: Old tests removed

File: server/src/test/java/com/defold/extender/AuthenticationTest.java (hunk 2 above)
Comment (JCash) [reply server/src/test/java/com/defold/extender/AuthenticationTest.java:34]: It's a bit strange though, as the sdk should be available?
http://d.defold.com/stable/

File: server/src/test/java/com/defold/extender/IntegrationTest.java [hunk 4]
```diff
@@ -115,13 +115,9 @@ public String toString() {
...
-
                 // // At 1.2.185, we updated to iOS 14.5, macOS 11.3 and XCode 12.5
                 new DefoldVersion("0a8d3e879724132afb18d47e0040c2034be07504", new Version(1, 2, 185), new String[] {"armv7-android", "armv7-ios", "arm64-ios", "x86_64-osx", "x86_64-linux", "x86_64-win32", "js-web", "wasm-web"}),
-                new DefoldVersion("d31d6397a72178541a5ef6e7ef2bed090d828f58", new Version(1, 2, 190), new String[] {"armv7-android", "armv7-ios", "arm64-ios", "x86_64-osx", "x86_64-linux", "x86_64-win32", "js-web", "wasm-web"}),
+                new DefoldVersion("d393bae6a361f86cf2263ab312c9b3cea45253ab", new Version(1, 2, 191), new String[] {"armv7-android", "armv7-ios", "arm64-ios", "x86_64-osx", "x86_64-linux", "x86_64-win32", "js-web", "wasm-web"}),
```
Comment (JCash) [inline server/src/test/java/com/defold/extender/IntegrationTest.java:120]: I would also add 1.3.1 (there are 3 releases after 1.2.191)

**Review by JCash — APPROVED** [review]

File: server/src/test/java/com/defold/extender/AuthenticationTest.java (hunk 2 above)
Comment (AGulev) [reply server/src/test/java/com/defold/extender/AuthenticationTest.java:34]: I agree, it is strange.
but I decided that the simplest way to fix that is to use new version (there are no big difference in this sense between 190 and 191)

**Pull request merged**